clients/unshaded/src/main/java/tachyon/client/block/RemoteBlockInStream.java

```diff
@@ -30,6 +33,13 @@
 public final class RemoteBlockInStream extends BufferedBlockInStream {
   /** The address of the worker to read the data from. */
   private final InetSocketAddress mLocation;
+  /** The returned lock id after acquiring the block lock */
+  private final Long mLockId;
+
+  /** Client to communicate with the remote worker. */
+  private final WorkerClient mWorkerClient;
```

Comment:
But the cached client will live across methods. And if for some reason, the caller forgot to close the stream, then the client will be leaked. So it's preferred to have the resource close as soon as its use finishes.